btcutil/hdkeychain/extendedkey.go

```diff
@@ -313,35 +312,66 @@ func (k *ExtendedKey) Derive(i uint32) (*ExtendedKey, error) {
 		// derive the final child key.
 		//
 		// childKey = parse256(Il) + parenKey
-		keyNum := new(big.Int).SetBytes(k.key)
-		ilNum.Add(ilNum, keyNum)
-		ilNum.Mod(ilNum, btcec.S256().N)
-		childKey = ilNum.Bytes()
+		var keyNum btcec.ModNScalar
+		if overflow := keyNum.SetByteSlice(k.key); overflow {
+			return nil, ErrInvalidChild
+		}
+
+		childKeyBytes := ilNum.Add(&keyNum).Bytes()
+		childKey = childKeyBytes[:]
+
+		// Strip leading zeroes from childKey, to match the expectation
```

Comment:
The only odd thing I encountered during this translation, ran into something similar when getting the tests to pass for this original PR. As is, `SetByteSlice` assumes that all leading zeroes are stripped off, which is why it's added in the modified `parseSig` operation. 

If this is removed, we fail `TestLeadingZero`.